front/lib/plans/subscription.ts

```diff
@@ -374,6 +375,17 @@ export async function maybeCancelInactiveTrials(
   }
 
   const { workspace } = subscription;
+
+  // This function can get called if the subscription is upgraded before the end of the trial.
+  // Ensure that the Stripe subscription still has a status set to `trialing`.
+  const stripeSubscription = await getStripeSubscription(stripeSubscriptionId);
+  if (!stripeSubscription || stripeSubscription.status !== "trialing") {
+    logger.info(
+      { action: "cancelling-trial", workspaceId: workspace.sId },
+      "Skipping cancel trial for active subscription."
```

Comment:
I would try to be more explicit, but that's a NIT feel free to merge as is. 


```
// Checking subscription is trialing on db
// Happens when ....
if subscriptionOnDB not trialing {
   log("Don't check for inactive trial. Subscription is still in trial on db");
   return
}

// Checking subscription is trialing on Stripe
// Happens when ...
if subscriptionOnStripe not trialing {
   log("Don't check for inactive trial. Subscription is still in trial on Stripe");
   return
}

Reply:
Seen IRL, will do some refactoring around this function in a follow up PR.